test/Constraints/diagnostics.swift

```diff
@@ -1190,10 +1189,9 @@ func rdar17170728() {
 // https://bugs.swift.org/browse/SR-5934 - failure to emit diagnostic for bad
 // generic constraints
 func elephant<T, U>(_: T) where T : Collection, T.Element == U, T.Element : Hashable {}
-// expected-note@-1 {{in call to function 'elephant'}}
 
 func platypus<T>(a: [T]) {
-    _ = elephant(a) // expected-error {{generic parameter 'U' could not be inferred}}
+    _ = elephant(a) // expected-error {{'([T]) -> ()' requires that 'T' conform to 'Hashable'}}
```

Comment:
Another case where we diagnose the type but not the decl -- shouldn't this say

```
function 'elephant' requires that 'T' conform to 'Hashable'
```

Reply:
Yes, that definitely would be great! I left that for last, but I think it shouldn't be hard to do because anchor point to either `DeclRefExpr` or `UnresolvedDot` which should have a kind/name attach to it.

Just trying to get protocol compositions diagnose correctly first...